core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/Streamlet.scala

```diff
@@ -16,13 +16,42 @@
 
 package cloudflow.streamlets
 
+import cloudflow.streamlets.descriptors.StreamletDescriptor
+import com.typesafe.config.Config
+import org.slf4j.LoggerFactory
+
 import scala.collection.immutable
 
-import com.typesafe.config.Config
+abstract class Streamlet[Context <: StreamletContext] {
+  @transient protected lazy val log = LoggerFactory.getLogger(getClass.getName)
```

Comment:
Looking at this now, I'm not so happy with slf4j dep at  this level, but I can fix that in a subsequent PR.